core/src/main/java/com/cprcrack/videowebview/VideoEnabledWebChromeClient.kt

```diff
@@ -0,0 +1,315 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2020 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package com.cprcrack.videowebview
+
+import android.media.MediaPlayer
+import android.view.SurfaceView
+import android.view.View
+import android.view.ViewGroup
+import android.webkit.WebChromeClient
+import android.widget.FrameLayout
+import android.widget.VideoView
+import org.kiwix.kiwixmobile.core.main.KiwixWebView
+
+/**
+ * This class serves as a WebChromeClient to be set to a WebView, allowing it to play video.
+ * Video will play differently depending on target API level (in-line, fullscreen, or both).
+ *
+ * It has been tested with the following video classes:
+ * - android.widget.VideoView (typically API level <11)
+ * - android.webkit.HTML5VideoFullScreen$VideoSurfaceView/VideoTextureView (typically API level
+ * 11-18)
+ * - com.android.org.chromium.content.browser.ContentVideoView$VideoSurfaceView (typically API level
+ * 19+)
+ *
+ * Important notes:
+ * - For API level 11+, android:hardwareAccelerated="true" must be set in the application manifest.
+ * - The invoking activity must call VideoEnabledWebChromeClient's onBackPressed() inside of its own
+ * onBackPressed().
+ * - Tested in Android API levels 8-19. Only tested on http://m.youtube.com.
+ *
+ * @author Cristian Perez (http://cpr.name)
+ */
+
+open class VideoEnabledWebChromeClient constructor() : WebChromeClient(),
+    MediaPlayer.OnCompletionListener,
+    MediaPlayer.OnPreparedListener, MediaPlayer.OnErrorListener {
+  constructor(
```

Comment:
I would perhaps recommend going through the android docs [tutorials](https://developer.android.com/kotlin/androidbasics) on kotlin, the kotlin documentation on [constructors](https://kotlinlang.org/docs/reference/classes.html#constructors) and when learning kotlin it never hurts to give the [koans](https://kotlinlang.org/docs/tutorials/koans.html) a try